eduvpn/nm.py

```diff
@@ -12,13 +15,40 @@
 
     gi.require_version('NM', '1.0')
     from gi.repository import NM, GLib  # type: ignore
-except (ImportError, ValueError) as e:
+except (ImportError, ValueError):
     logger.warning("Network Manager not available")
     NM = None
 
 from eduvpn.storage import set_uuid, get_uuid, write_config
 from eduvpn.utils import get_logger
 
+
+class ConnectionState(Flag):
+    UNKNOWN = 0  # NM.VpnConnectionState.UNKNOWN
+    PREPARE = 1  # NM.VpnConnectionState.PREPARE
+    NEED_AUTH = 2  # NM.VpnConnectionState.NEED_AUTH
+    CONNECT = 3  # NM.VpnConnectionState.CONNECT
+    IP_CONFIG_GET = 4  # NM.VpnConnectionState.IP_CONFIG_GET
+    ACTIVATED = 5  # NM.VpnConnectionState.ACTIVATED
+    FAILED = 6  # NM.VpnConnectionState.FAILED
+    DISCONNECTED = 7  # NM.VpnConnectionState.DISCONNECTED
+
+
+class ConnectionStateReason(Flag):
+    UNKNOWN = 0  # NM.VpnConnectionStateReason.UNKNOWN
+    NONE = 1  # NM.VpnConnectionStateReason.NONE
+    USER_DISCONNECTED = 2  # NM.VpnConnectionStateReason.USER_DISCONNECTED
+    DEVICE_DISCONNECTED = 3  # NM.VpnConnectionStateReason.DEVICE_DISCONNECTED
+    SERVICE_STOPPED = 4  # NM.VpnConnectionStateReason.SERVICE_STOPPED
+    IP_CONFIG_INVALID = 5  # NM.VpnConnectionStateReason.IP_CONFIG_INVALID
+    CONNECT_TIMEOUT = 6  # NM.VpnConnectionStateReason.CONNECT_TIMEOUT
+    SERVICE_START_TIMEOUT = 7  # NM.VpnConnectionStateReason.SERVICE_START_TIMEOUT
+    SERVICE_START_FAILED = 8  # NM.VpnConnectionStateReason.SERVICE_START_FAILED
+    NO_SECRETS = 9  # NM.VpnConnectionStateReason.NO_SECRETS
+    LOGIN_FAILED = 10  # NM.VpnConnectionStateReason.LOGIN_FAILED
+    CONNECTION_REMOVED = 11  # NM.VpnConnectionStateReason.CONNECTION_REMOVED
```

Comment:
I dont fully understand why you replicate these numbers here? they are available in network manager right?

Reply:
The docker tests fail because they don't have NM available and then NM above is None:

______________________ ERROR collecting tests/test_cli.py ______________________
tests/test_cli.py:2: in <module>
    from eduvpn.cli import interactive, start
eduvpn/cli.py:11: in <module>
    from eduvpn.nm import activate_connection, deactivate_connection
eduvpn/nm.py:26: in <module>
    class ConnectionState(Flag):
eduvpn/nm.py:27: in ConnectionState
    UNKNOWN = NM.VpnConnectionState.UNKNOWN
E   AttributeError: 'NoneType' object has no attribute 'VpnConnectionState'

Do you have a better way in mind?

Reply:
are you sure? Maybe `gir1.2-nm-1.0` is then not installed?

This works fine for me:
```

FROM ubuntu:18.04                                                                                                       
RUN apt-get update                                                                                                      
RUN apt-get -y install gir1.2-nm-1.0 python3-gi                                                                         
RUN python3 -c "import gi;gi.require_version('NM', '1.0');from gi.repository import NM;print(NM.VpnConnectionState.UNKNOWN)"
```

Reply:
Thx, that works. Also found the correct package for rpm based distros.